### DISCUSSION TITLE
xml2cpp : how to add const to method definition?

### DISCUSSION THREAD

**Discussion by Arniiiii** [Kistler-Group/sdbus-cpp#349] [Q&A]
e.g. how to get from xml such code:
```
namespace ... {

class ... {

// a method
virtual return_type a_method (...) const = 0;

// a property

virtual return_type getter_property() const = 0;

}


}

```

Right now, if try access="read" or access="readwrite" in an xml property tag, it's just adds a method, but not const.

Is there an annotation to get const method or something like ?

**Comment by sangelovic** [comment]
No. AFAIK, there is no support for `const` in D-Bus XML specification. The only way I can think of would be that we introduce that as a sdbus-c++-specific extension to the XML format, in a form of an annotation, for example.

Bear in mind however, when you introspect such a service with D-Bus introspection tools, sdbus-c++-specific extensions will be omitted.

**Reply by Arniiiii** [reply]
Thanks! I would be pleased to see such annotation, because some part of my code is non-const, though, if check their implementation, they should be const.
